grafana/grafana
# Elasticsearch: Remove browser access mode

Fixes https://github.com/grafana/grafana/issues/12757

in the Elasticsearch datasource configuration, you can choose the access mode, `browser` or `server`. the `browser` mode was deprecated in grafana7.4.0, and now it has been removed.

to be the least surprising and confusing, i chose the solution where we do not change anything automatically, except browser-mode stops working. in such a situation we explain to the user why it is not working and how to fix the problem. the user has to change the configuration.

in the documentation i removed all the explanations for this setting because now there are only two possible settings: works & does_not_work.

how to test:
- setup an elasticsearch datasource.
- switch the access-mode to `browser`
- you should see an error message at the top
- press [save&test], you should get the same error-message
- now go to explore-mode, and try to query this datasource
- you should get the same error-message again
- now go back to datasource-config
- switch the access-mode to `server`
- the error-message should vanish
- press [save&test], you should get no error-message
- go to explore, run a query, you should get no error-message

# Release notice breaking change

In Elasticsearch, browser access mode was deprecated in grafana 7.4.0 and removed in 9.0.0. If you used this mode, please switch to server access mode on the datasource configuration page.
